utils/src/main/java/uk/gov/service/payments/commons/utils/prometheus/EcsTaskMetadataVersion4.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.service.payments.commons.utils.prometheus;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Optional;
+
+/**
+ * Extracts relevant information from the <a href="https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html">AWS ECS task metadata endpoint version 4</a>.
+ */
+public class EcsTaskMetadataVersion4 {
+    
+    private final JSONObject ecsTaskMetadataJson;
+    private final String clusterName;
+    private final String accountName;
+    private final String ecsTaskId;
+    private final String containerImageTag;
+    private final String serviceName;
+    private final String instanceLabel;
+
+    /**
+     * Will throw a runtime exception if JSON is not valid.
+     * @param ecsTaskMetadataJson A valid <a href="https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html#task-metadata-endpoint-v4-response">Task metadata JSON</a>
+     */
+    public EcsTaskMetadataVersion4(String ecsTaskMetadataJson) {
+        this.ecsTaskMetadataJson = new JSONObject(ecsTaskMetadataJson);
+        
+        String cluster = this.ecsTaskMetadataJson.getString("Cluster");
+        this.clusterName = cluster.substring(cluster.lastIndexOf("/") + 1);
+
+        this.accountName = clusterName.substring(0, clusterName.indexOf("-"));
+        
+        String taskARN = this.ecsTaskMetadataJson.getString("TaskARN");
+        this.ecsTaskId = taskARN.substring(taskARN.lastIndexOf("/") + 1);
+
+        JSONObject container = this.ecsTaskMetadataJson.getJSONArray("Containers").getJSONObject(0);
```

Comment:
is object 0 always the java container?

Reply:
Object 0 is just getting the [first json structure from the "Containers" list](https://github.com/alphagov/pay-java-commons/pull/691/files#diff-b2982d5a0905d509ec74e9247fe1e9bf0fa83a314d4732153b927a7ab0302212R16).

Reply:
There should always be at least 1. (We have 3 in that test json because there's connector, nginx and telegraf. Arguably nginx and telegraf don't have to be there - I think I'll delete.)

Reply:
yes, but I'm asking if the order is always guaranteed when requesting from the endpoint given we have three containers running in each task.

Reply:
Ahh, good question/spot. I will filter the relevant json. But this would mean passing in the service we're interested in (in this case "connector") to the constructor.